plugins/optimization-detective/detect.js

```diff
@@ -361,39 +381,80 @@ export default async function detect( {
 		log( 'URL metrics:', urlMetrics );
 	}
 
-	// Yield to main before sending data to server to further break up task.
-	await new Promise( ( resolve ) => {
-		setTimeout( resolve, 0 );
-	} );
+	async function sendData() {
+		// Data likely already sent.
+		if (
+			! webVitalsMetrics.LCP.length &&
+			! webVitalsMetrics.CLS.length &&
+			! webVitalsMetrics.INP.length &&
+			! webVitalsMetrics.TTFB.length
+		) {
+			return;
+		}
 
-	try {
-		const response = await fetch( restApiEndpoint, {
-			method: 'POST',
-			headers: {
-				'Content-Type': 'application/json',
-				'X-WP-Nonce': restApiNonce,
-			},
-			body: JSON.stringify( urlMetrics ),
-		} );
+		for ( const [ webVital, metric ] of Object.entries(
+			webVitalsMetrics
+		) ) {
+			if ( ! metric.length ) {
+				continue;
+			}
 
-		if ( response.status === 200 ) {
-			setStorageLock( getCurrentTime() );
+			urlMetrics.webVitals[ webVital ] = metric[ 0 ].value;
 		}
 
-		if ( isDebug ) {
-			const body = await response.json();
+		log( 'Prepared url metrics', urlMetrics );
+
+		try {
+			const response = await fetch( restApiEndpoint, {
+				method: 'POST',
+				headers: {
+					'Content-Type': 'application/json',
+					'X-WP-Nonce': restApiNonce,
+				},
+				body: JSON.stringify( urlMetrics ),
+				keepalive: true,
+			} );
+
 			if ( response.status === 200 ) {
-				log( 'Response:', body );
-			} else {
-				error( 'Failure:', body );
+				setStorageLock( getCurrentTime() );
+			}
+
+			if ( isDebug ) {
+				const body = await response.json();
+				if ( response.status === 200 ) {
+					log( 'Response:', body );
+				} else {
+					error( 'Failure:', body );
+				}
+			}
+		} catch ( err ) {
+			if ( isDebug ) {
+				error( err );
 			}
 		}
-	} catch ( err ) {
-		if ( isDebug ) {
-			error( err );
-		}
+
+		// Clean up.
+		breadcrumbedElementsMap.clear();
+
+		webVitalsMetrics.LCP = [];
+		webVitalsMetrics.CLS = [];
+		webVitalsMetrics.INP = [];
+		webVitalsMetrics.TTFB = [];
 	}
 
-	// Clean up.
-	breadcrumbedElementsMap.clear();
+	log( 'Adding event listeners for backgrounding and unloading.' );
```

Comment:
```suggestion
	if ( isDebug ) {
		log( 'Adding event listeners for backgrounding and unloading.' );
	}
```

Reply:
Meh, I have a lot of this `if( isDebug ) { log(...) }` code in this file. Probably should introduce a `debugLog`, `debugWarn`, and `debugError()` functions that remove this repetition.